File: ruby/sum-of-multiples/sum_of_multiples.rb
module BookKeeping
  VERSION = 1
end

class SumOfMultiples
  def initialize(*nums)
    @multiples = nums
  end

  # find multiples of #{@multples} up to #{limit}
  def to(limit)
    result = []
    @multiples.each do |mult|
      (1...limit).each do |i|
        result << i if (i % mult).zero? && result.include?(i) == false
      end
    end
    result.sum
  end
end

File: csharp/sum-of-multiples/SumOfMultiples.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class SumOfMultiples
{
    public static int Sum(IEnumerable<int> multiples, int max)
    {
        HashSet<int> result = new HashSet<int>();
        
        foreach (int mult in multiples) 
        {
            for (int i = mult; i < max; i++) 
            {
                if (i % mult == 0)
                {
                    result.Add(i); 
                }
            }
        }
        return result.Sum();
    }
}